packages/syft/src/syft/node/node.py

```diff
@@ -1243,21 +1243,76 @@ def add_queueitem_to_queue(
             return result
         return job
 
+    def _can_enqueue_user_code_job(
```

Comment:
I think this function might be problematic in some cases. In the case of nested jobs we follow the logic that: once you can execute the outer job, you will automatically get permissions to execute inner jobs. This information (that you come from an outer job), will be added to the context. I am not sure if we currently have a test that tests this properly, but this might break that

Reply:
Decided to keep it like this for now, context: https://openmined.slack.com/archives/C05TVM4RZKQ/p1705646834228349?thread_ts=1705593124.565069&cid=C05TVM4RZKQ